### PR TITLE
Fix a significant performance regression introduced in v8.1.0

### PR DESCRIPTION
**Pull request by mar10**
❓ **What kind of change does this PR introduce?**
  - [ ] 🐞 bug fix
  - [X] 🐣 feature
  - [ ] 📋 docs update
  - [ ] 📋 tests/coverage improvement
  - [ ] 📋 refactoring
  - [ ] 💥 other

📋 **What is the related issue number (starting with `#`)**

Hotfixes #305 

❓ **What is the current behavior?** (You can also link to an open issue here)

Starting with v8.1, Cheroot's performance dropped in certain scenarios. See #305 for details.

❓ **What is the new behavior (if this is a feature change)?**

The suggested change restores (and tops) the performance again (see benchmarks in #305)

📋 **Other information**:

N/A

📋 **Checklist**:

  - [x] I think the code is well written
  - [x] I wrote [good commit messages][1]
  - [ ] I have [squashed related commits together][2] after the changes have been approved
  - [ ] Unit tests for the changes exist
  - [ ] Integration tests for the changes exist (if applicable)
  - [x] I used the same coding conventions as the rest of the project
  - [ ] The new code doesn't generate linter offenses
  - [ ] Documentation reflects the changes
  - [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

---
This change is [Reviewable](https://reviewable.io/reviews/cherrypy/cheroot/308)

### CODE REVIEW

File: cheroot/connections.py [hunk 1]
```diff
@@ -136,9 +136,11 @@ def get_conn(self, server_socket):
         try:
             for fno in socket_dict:
                 self._selector.register(fno, selectors.EVENT_READ)
+            # The timeout value impacts performance and should be carefully
+            # chosen (issue #305)
```
Comment (webknjaz) [inline cheroot/connections.py:140]: I prefer having a complete URL in the comment (this is friendlier to envs outside of GH)
```suggestion
            # chosen. Ref:
            # github.com/cherrypy/cheroot/issues/305#issuecomment-663985165
```

**Comment by webknjaz** [comment]
@mar10, thanks for taking a stab at it! Mind cleaning up the branch a bit?

**Comment by webknjaz** [comment]
cc @tommilligan @liamstask @kmduk @timurgen @jakeczyz @liamstask @hardikmodha @manthey @kavehtp @andebor @the-allanc

**Comment by liamstask** [comment]
it looks to me that this could be related to the fact that, currently in `server.tick()`, only one request can be added per tick even if several are available to be processed in `ConnectionManager.connections`.

in this case, since `ConnectionManager` does not register connections previously marked as read-ready with the selector, it's possible that _none_ of the remaining connections in the selector are active, meaning the server is forced to wait for the full timeout duration each time `tick()` is called.

fundamentally, the value of this timeout should not really be particularly sensitive, since the selector should wake up once connections are ready to be processed, regardless of the timeout.

**Comment by webknjaz** [comment]
Thanks for the insight, @liamstask! What you say sounds reasonable, I haven't considered it...
Looks like improving this would require more fundamental changes and the hotfix of changing the timeout would be acceptable short-term. WDYT?

**Comment by liamstask** [comment]
yep - that seems like an OK short term hotfix to me 👍

**Comment by webknjaz** [comment]
Hotfix released as v8.4.2: https://pypi.org/project/cheroot/8.4.2/ (https://travis-ci.com/github/cherrypy/cheroot/jobs/365552834#L519)

**Pull request merged**